doc/migrations/v0.43-v0.44.md

```diff
@@ -0,0 +1,117 @@
+# Migrating to libp2p@44 <!-- omit in toc -->
+
+A migration guide for refactoring your application code from libp2p v0.43.x to v0.44.0.
+
+## Table of Contents <!-- omit in toc -->
+
+- [Connection Manager](#connection-manager)
+- [Connection Gater](#connection-gater)
+
+## Connection Manager
+
+The connection manager has been refactored to be simpler and faster.  The changes
+are mostly internal but some configuration options have changed.
+
+**Before**
+
+```js
+import { createLibp2p } from 'libp2p'
+
+const node = await createLibp2p({
+  connectionManager: {
+    // this option has been removed as auto dial now runs in response to peers
+    // disconnecting and is no longer on a timer
+    autoDialInterval: 0,
+
+    // this has been renamed to `maxPeerAddrsToDial`
+    maxAddrsToDial: 20,
+
+    // this has been renamed to `maxParallelDialsPerPeer`
+    maxDialsPerPeer: 20,
+
+    // these should now be ranges expressed as MultiaddrFilters
+    allow: [
+      '/ip4/0.0.0.0/tcp/123'
+    ],
+
+    // these should now be ranges expressed as MultiaddrFilters
+    deny: [
+      '/ip4/0.0.0.0/tcp/123'
+    ]
```

Comment:
Can we get a "allow peer" that allows any appropriate transports given a specific peerId?

I was trying to do that the other day in kubo and realized i needed a full multiaddr. I was trying to grant full-access to/from my helia node and couldn't figure out how to do it within an hour or so and i gave up.

It would be great to be able to do that in js-libp2p

Reply:
I'm a little confused, access is allowed by default, it's only disallowed if a configured connection gater denys it or you're at `maxConnections` already.

The `allow` list above enables specific networks to breach connection limits - a DoS mitigation for when your node is being eclipsed.

Maybe you can open an issue on the helia repo with a repro case?

Reply:
sure. I will re-open if I run into it.. essentially what I was trying to do was prefer my local node to others, and to allow connections to that one no matter what. it sounds like that may not be necessary